pkg/invariants/impl.go

```diff
@@ -85,77 +82,73 @@ func (r *invariantRegistry) StartCollection(ctx context.Context, adminRESTConfig
 	return junits, utilerrors.NewAggregate(errs)
 }
 
-func (r *invariantRegistry) CollectData(ctx context.Context) (monitorapi.Intervals, []*junitapi.JUnitTestCase, error) {
+func (r *invariantRegistry) CollectData(ctx context.Context, storageDir string, beginning, end time.Time) (monitorapi.Intervals, []*junitapi.JUnitTestCase, error) {
 	intervals := monitorapi.Intervals{}
 	junits := []*junitapi.JUnitTestCase{}
 	errs := []error{}
 
-	r.collectData.Do(func() {
-		for _, invariant := range r.invariantTests {
-			testName := fmt.Sprintf("jira/%q invariant test %v collection", invariant.jiraComponent, invariant.name)
-
-			start := time.Now()
-			localIntervals, localJunits, err := invariant.invariantTest.CollectData(ctx)
-			junits = append(junits, localJunits...)
-			intervals = append(intervals, localIntervals...)
-			end := time.Now()
-			duration := end.Sub(start)
-			if err != nil {
-				errs = append(errs, err)
-				junits = append(junits, &junitapi.JUnitTestCase{
-					Name:     testName,
-					Duration: duration.Seconds(),
-					FailureOutput: &junitapi.FailureOutput{
-						Output: fmt.Sprintf("failed during collection\n%v", err),
-					},
-					SystemOut: fmt.Sprintf("failed during collection\n%v", err),
-				})
-				continue
-			}
+	for _, invariant := range r.invariantTests {
+		testName := fmt.Sprintf("jira/%q invariant test %v collection", invariant.jiraComponent, invariant.name)
```

Comment:
Why the different syntax here from our normal [component] prefix?

Reply:
updated in #28104